astropy/visualization/lupton_rgb.py

```diff
@@ -0,0 +1,469 @@
+
+    def make_rgb_image(self, image_r=None, image_g=None, image_b=None,
+                       x_size=None, y_size=None, rescale=None):
+        """
+        Convert 3 arrays, image_r, image_g, and image_b into an 8-bit RGB image.
+
+        Parameters
+        ----------
+        image_r : `~numpy.ndarray`, optional
+            Image to map to red (if None, use the image passed to the
+            constructor).
+        image_g : `~numpy.ndarray`, optional
+            Image to map to green (if None, use image_r).
+        image_b : `~numpy.ndarray`, optional
+            Image to map to blue (if None, use image_r).
+        x_size : int, optional
+            Desired width of RGB image (or None).  If y_size is None, preserve
+            aspect ratio.
+        y_size : int, optional
+            Desired height of RGB image (or None).
+        rescale : float, optional
+            Make size of output image rescale*size of the input image.
+            Cannot be specified if x_size or y_size are given.
+
+        Returns
+        -------
+        RGBimage : `~numpy.ndarray`
+            RGB (integer, 8-bits per channel) color image as an NxNx3 numpy array.
+        """
+        if image_r is None:
+            if self._image is None:
+                raise RuntimeError("you must provide an image or pass one "
+                                   "to the constructor.")
+            image_r = self._image
+        else:
+            image_r = np.asarray(image_r)
+
+        if image_g is None:
+            image_g = image_r
+        else:
+            image_g = np.asarray(image_g)
+
+        if image_b is None:
+            image_b = image_r
+        else:
+            image_b = np.asarray(image_b)
+
+        if x_size is not None or y_size is not None:
+            if rescale is not None:
+                raise ValueError("you may not specify a size and rescale.")
+            h, w = image_r.shape
+            if y_size is None:
+                y_size = int(x_size*h/float(w) + 0.5)
+            elif x_size is None:
+                x_size = int(y_size*w/float(h) + 0.5)
+
+            # need to cast to int when passing tuple to imresize.
+            size = (int(y_size), int(x_size))  # n.b. y, x order for scipy
+        elif rescale is not None:
+            size = float(rescale)  # a float is intepreted as a percentage
+        else:
+            size = None
+
+        if size is not None:
+            if not HAVE_SCIPY_MISC:
+                raise RuntimeError("unable to rescale as scipy.misc.imresize "
+                                   "is unavailable.")
+
+            image_r = scipy.misc.imresize(image_r, size, interp='bilinear',
+                                          mode='F')
+            image_g = scipy.misc.imresize(image_g, size, interp='bilinear',
+                                          mode='F')
+            image_b = scipy.misc.imresize(image_b, size, interp='bilinear',
+                                          mode='F')
```

Comment:
I think this kind of thing should be done outside of `make_lupton_rgb`. For easier maintenance and evolution, the code here should concentrate only on doing the optimal combination of the images, other things like resizing (or reprojecting) can be done upstream, and possibly with different methods.